cluster/kubernetes/release.go

```diff
@@ -49,16 +58,63 @@ func (c *Kubectl) connectArgs() []string {
 	return args
 }
 
-func (c *Kubectl) kubectlCommand(args ...string) *exec.Cmd {
-	cmd := exec.Command(c.exe, append(c.connectArgs(), args...)...)
-	cmd.Stdout = c.stdout
-	cmd.Stderr = c.stderr
-	return cmd
+func (c *Kubectl) execute(logger log.Logger, errs cluster.SyncError) {
+	defer c.changeSet.clear()
+
+	for _, cmd := range cmds {
+		defaultSet := newExecuteSet(cmd, "--namespace", "default")
+		otherSet := newExecuteSet(cmd)
+
+		for _, obj := range c.objs[cmd] {
+			set := defaultSet
+			if !obj.hasDefaultNamespace() {
+				set = otherSet
+			}
+			set.stage(obj)
+		}
+
+		c.exec(logger, defaultSet, cmd, errs)
```

Comment:
If you always apply things without namespaces first, maybe you can get rid of the sorting of namespace resources to the front, in `sync/sync.go`?

Reply:
(in fact I would prefer that, because it is particular to Kubernetes, which that package is not supposed to be ..)

Reply:
Done.